### PR TITLE
iOS-978 Core | Refactoring ObjectIconImageView

### PR DESCRIPTION
**Pull request by mgolovko**
https://github.com/anyproto/anytype-swift/assets/2483784/ce58db20-fb34-4ee8-b1ac-1a1eeae3178e

### CODE REVIEW

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/IconView.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import Foundation
+import SwiftUI
+
+struct IconView: View {
+    
+    let icon: Icon?
+    
+    @State private var task: Task<Void, Never>?
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/IconView.swift:8]: do we really need to set task like `@State` here?

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/Painters/Common/Container/SquareDynamicIconPainter.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        
+        context.restoreGState()
+    }
+    
+    // MARK: - Private
+    
+    private func drawBackground(bounds: CGRect, context: CGContext, iconContext: IconContext) {
+        let side = min(bounds.size.width, bounds.size.height)
+        
+        let path = UIBezierPath(roundedRect: bounds, cornerRadius: side * (1/12)).cgPath
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/Painters/Common/Container/SquareDynamicIconPainter.swift:39]: why `1/12`?

**Comment by joe-pusya** [comment]
Stunning 🔥

File: Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/Painters/Common/Container/SquareDynamicIconPainter.swift (hunk 2 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/Common/SwiftUI/IconView/Painters/Common/Container/SquareDynamicIconPainter.swift:39]: Corner radius depends from size. Maked a formula from design

**Pull request merged**